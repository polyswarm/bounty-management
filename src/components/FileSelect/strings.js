const strings = {
  remove: 'x'
};

export default strings;
