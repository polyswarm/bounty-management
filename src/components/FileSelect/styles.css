.FileProgress {
  border-bottom: 2px #eeeeee solid;
  box-sizing: border-box;
  background-clip: border-box;
  height: 48px;
  line-height: 46px;
  margin: 0 10px;
  width: calc(100% - 20px);
}

.FileProgress-Name {
  float: left;
  margin: 0;
}

.FileProgress-Remove {
  float: right;
}
