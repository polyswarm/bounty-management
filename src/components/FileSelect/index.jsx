import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import RemoveButton from '../RemoveButton';
// Component imports
import strings from './strings';
import './styles.css';

class FileSelect extends Component {
  constructor(props) {
    super(props);
    this.onClickHandler = this.onClickHandler.bind(this);
  }

  render () {
    const { props: { file } } = this;
    return(
      <div className='FileProgress'>
        <p className='FileProgress-Name'>{file.name}</p>
        <span className='FileProgress-Remove'>
          <RemoveButton onClick={this.onClickHandler}>
            {strings.remove}
          </RemoveButton>
        </span>
      </div>
    );
  }

  onClickHandler() {
    const { props: { remove } } = this;
    if (remove) {
      remove();
    }
  }
}

FileSelect.proptypes = {
  file: PropTypes.string.isRequired,
  remove: PropTypes.func.isRequired,
};

export default FileSelect;
