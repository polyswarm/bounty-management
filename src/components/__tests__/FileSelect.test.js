import React from 'react';
import {render} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import FileSelect from '../FileSelect';

it('throws on render without files', () => {
  expect(() => {
    render(<FileSelect />);
  }).toThrow();
});

it('renders without crashing', () => {
  const file = {
    name: 'asdf'
  };
  const wrapper = render(<FileSelect file={file} />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

if('calls the mock function when I click close', () => {
  const remove = jest.fn();
  const file = {
    name: 'asdf'
  };
  const wrapper = render(<FileSelect file={file} remove={remove} />);
  wrapper.find('button').simulate('click');
  expect(remove).toHaveBeenCalled();
});
