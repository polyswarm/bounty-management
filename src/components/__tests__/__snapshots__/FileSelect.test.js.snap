// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="FileProgress"
>
  <p
    class="FileProgress-Name"
  >
    asdf
  </p>
  <span
    class="FileProgress-Remove"
  >
    <button
      class="Remove-Button"
    >
      x
    </button>
  </span>
</div>
`;
